### ISSUE TITLE
index.html template rendered as text when requesting "/"

### ISSUE THREAD

**Issue by ciarand** [mholt/caddy#27]
This is probably just an ordering issue somewhere, but it had me scratching my head for a couple of minutes.

Steps to reproduce:

1. Create the following files in a new directory:

    index.html:
    ```html
    {{/* super secret information goes here */}}
    Public information goes here
    ```
    Caddyfile:
    ```
    localhost:2020
    template / .html
    ```

2. Run `caddy`
3. Visit [http://localhost:2020/](http://localhost:2020/) and [http://localhost:2020/index.html](http://localhost:2020/index.html) and compare the results

Expected results:
- Both display only "Public information goes here"

Actual results:
- http://localhost:2020/ displays the source of the file, starting with `{{/*`

**Comment by mholt** [comment]
I bet the bug is [here](https://github.com/mholt/caddy/blob/60beddf6c8186ef06e5a50d03cf0fdb9d811be59/middleware/templates/templates.go#L38-L41):

```go
reqExt := path.Ext(r.URL.Path)

for _, ext := range rule.Extensions {
	if reqExt == ext {
```

"/" doesn't have a ".html" extension, so the comparison fails and the template doesn't get executed. Should be a simple matter of checking if the path is a directory first, and if so, if an index file exists.

Although, more middlewares are going to run into this issue. Right now the list of index files is actually defined in the [browse middleware](https://github.com/mholt/caddy/blob/60beddf6c8186ef06e5a50d03cf0fdb9d811be59/middleware/browse/browse.go#L68-L73) which the core fileServer references. But maybe this isn't the best way to do it.

Perhaps each middleware could have an `IndexFiles []string` field that the _application_ sets. To make this work well, I need to finish refactoring the config package and get those generator (`New()`) functions out of the middleware packages.

Okay - in summary: the simple fix is to just have the template middleware look for `index.<extension>` if the request is to a directory. That could hold us over for now, maybe?

**Comment by ciarand** [comment]
Yeah, that sounds reasonable. Though this was enough to hold me over for now:

```Caddyfile
rewrite / /index.html
```

**Comment by mholt** [comment]
@ciarand This fix will go out at the next release.

**Comment by ciarand** [comment]
Sounds great, thanks @mholt!
